src/adapters/profile.ts

```diff
@@ -51,145 +37,44 @@ class Profile {
-  ): Promise<UserProfile> {
-    const res = await fetch(
-      `${API_BASE_URL}/profiles?guild_id=${guildId}&user_id=${userId}`,
-      {
-        method: "GET",
-      }
-    )
-    if (res.status !== 200) {
-      throw new Error(
-        `failed to get profile of user ${userId} - guild ${guildId}`
-      )
-    }
-
-    const json = await res.json()
-    if (json.error !== undefined) {
-      throw new Error(json.error)
-    }
-    return json.data
-  }
-
-  public async getPodTownUser(discordId: string): Promise<PodTownUser | null> {
-    const resp = await fetch(
-      `${PT_API_SERVER_HOST}/api/v1/user/info?discord_id=${discordId}`,
-      {
-        method: "GET",
-        headers: {
-          "Content-Type": "application/json",
-          // Authorization: `Bearer ${NEKO_SECRET}`,
-        },
-      }
+  public async getUserProfile(guildId: string, userId: string) {
+    return await this.jsonFetch<UserProfileResponse>(
+      `${API_BASE_URL}/profiles?guild_id=${guildId}&user_id=${userId}`
     )
-    const json = await resp.json()
-    switch (json.error) {
-      case "unverified user":
-        return {
-          ...json.user,
-          is_verified: false,
-        }
-      case undefined:
-        return {
-          ...json.user,
-          is_verified: true,
-        }
-      default:
-        return null
-    }
   }
 
   public async getUserNFTCollection(params: {
     userAddress: string
     page?: number
     size?: number
-  }): Promise<GetUserNFTCollectionResponse> {
+  }) {
     const { userAddress, page = 0, size = 50 } = params
-    const res = await fetch(
-      `${INDEXER_API_BASE_URL}/${userAddress}/collection?page=${page}&size=${size}`,
-      {
-        method: "GET",
-      }
+    return await this.jsonFetch<GetUserNFTCollectionResponse>(
+      `${INDEXER_API_BASE_URL}/${userAddress}/collection?page=${page}&size=${size}`
     )
-    if (res.status !== 200) {
-      throw new Error(
-        `failed to get collections of user address ${userAddress} `
-      )
-    }
-
-    const json = await res.json()
-    if (json.error !== undefined) {
-      throw new Error(json.error)
-    }
-    return json
   }
 
   public async getUserNFT(params: {
     userAddress: string
     collectionAddress?: string
     page?: number
     size?: number
-  }): Promise<GetUserNFTResponse> {
+  }) {
     const { userAddress, collectionAddress, page = 0, size = 50 } = params
     let url = `${INDEXER_API_BASE_URL}/${userAddress}/nft?page=${page}&size=${size}`
     if (collectionAddress) {
       url = `${url}&collection_addresses=${collectionAddress}`
     }
-    const res = await fetch(url, {
-      method: "GET",
-    })
-    if (res.status !== 200) {
-      throw new Error(`failed to get nfts of user address ${userAddress} `)
-    }
-
-    const json = await res.json()
-    if (json.error !== undefined) {
-      throw new Error(json.error)
-    }
-    return json
+    return await this.jsonFetch<GetUserNFTsResponse>(url)
```

Comment:
```suggestion
    return await this.jsonFetch<GetUserNFTsResponse>(`${INDEXER_API_BASE_URL}/${userAddress}/nft`, {
        query: {
            page,
            size,
            collectionAddresses
        }
    })
```

Chỗ này ông nhớ sửa lại tên param `collectionAddress` -> `collectionAddresses` để có thể dùng shorthand property syntax của ES, với lại nếu `collectionAddresses` mà = undefined thì nó cũng sẽ tự động bị skip khi convert qua query string thôi nên không cần logic check if luôn